bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -1553,87 +1558,97 @@ public void executeSequenceCommand(@Nullable Device device, String command,
     }
 
     private void executeSequenceNode(Device[] devices, JsonObject nodeToExecute) {
-        if (devices.length == 1 && groups.values().stream().anyMatch(queueObject -> queueObject.queueRunning.get())
-                || devices.length > 1
-                        && singles.values().stream().anyMatch(queueObject -> queueObject.queueRunning.get())) {
-            if (singleGroupTimer != null) {
-                singleGroupTimer.cancel(true);
-                singleGroupTimer = null;
+        QueueObject queueObject = new QueueObject();
+        queueObject.devices = devices;
+        queueObject.nodeToExecute = nodeToExecute;
+        String serial = "";
+        for (Device device : devices) {
+            if (!this.devices.containsKey(device.serialNumber)) {
+                this.devices.put(device.serialNumber, new LinkedBlockingQueue<>());
             }
```

Comment:
`computeIfAbsent` would simplify this a bit.
```suggestion
            this.devices.computeIfAbsent(device.serialNumber, k -> new LinkedBlockingQueue<>()).offer(queueObject);
```